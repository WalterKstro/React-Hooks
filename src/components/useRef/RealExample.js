import MultiplesCustomHooks from '../examples/MultiplesCustomHooks'
import useToogle from '../../hooks/useToogle'

import Switcher from '../Switch/Switcher'

const RealExample = () => {
    const {state, handleToogle} = useToogle()
    return (
        <>
         { state && <MultiplesCustomHooks/>}
          <hr/>
          <Switcher handleToogle={handleToogle}/>
        </>
    )
}

export default RealExample
