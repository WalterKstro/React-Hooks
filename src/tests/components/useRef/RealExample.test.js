import { shallow } from "enzyme"
import RealExample from "../../../components/useRef/RealExample"

describe('Testing component <RealExample/>', () => {
    test('should show a snaptshot of component <RealExample/>', () => {
        const rendered = shallow(<RealExample/>)
        expect(rendered).toMatchSnapshot() 
    })

    test('should show or hidden the component <RealExample/>', () => {
        const rendered = shallow(<RealExample/>)
        const isComponent = rendered.find('MultiplesCustomHooks').exists()
        const toogle = rendered.find('Switcher').exists()

        expect(isComponent).toBe(true)
        expect(toogle).toBe(true)
    })
    
    
})
