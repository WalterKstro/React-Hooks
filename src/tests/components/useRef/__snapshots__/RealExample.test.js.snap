// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component <RealExample/> should show a snaptshot of component <RealExample/> 1`] = `
<Fragment>
  <MultiplesCustomHooks />
  <hr />
  <Switcher
    handleToogle={[Function]}
  />
</Fragment>
`;
